lib/pages/settings/view/settings_page.dart

```diff
@@ -96,6 +111,36 @@ class SettingsPageState extends State<SettingsPage> {
                                 context: context),
                           ),
                         ),
+                        const Divider(),
+                        FlatButton(
+                          key: const ValueKey('ask_permissions'),
+                          onPressed: () => {
+                            if (!authProvider.isVerifiedFromCache &&
+                                !authProvider.isAnonymous)
+                              {
+                                AppToast.show(S
+                                    .of(context)
+                                    .messageEmailNotVerifiedAskPermissions)
+                              }
+                            else if (!authProvider.isAnonymous)
+                              {
+                                Navigator.of(context).push(
+                                    MaterialPageRoute<AskPermissions>(
+                                        builder: (context) =>
+                                            AskPermissions())),
+                              }
+                          },
+                          child: Text(S.of(context).labelAskPermissions,
+                              textAlign: TextAlign.center,
+                              style: authProvider.isAnonymous
```

Comment:
Ahh I see now what you did here. You can make the text greyed out when the user is not verified as well, and either not show it (like I suggested above) when the user is anonymous or show a different toast to ask the user to authenticate. Don't just grey it out, it can be unclear why an option like this is not available if you don't show a message.

Also, just so you know (it doesn't apply here) - if you set `onPressed` to `null` (or don't specify it) in an `IconButton`, `FlatButton` and similar tappable widgets, Flutter will automatically give them the disabled colour.